src/Diagnostics.DataProviders/KustoClient.cs

```diff
@@ -95,6 +95,13 @@ public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, str
 
                 if (!responseMsg.IsSuccessStatusCode)
                 {
+                    if ((int)responseMsg.StatusCode == 400)
```

Comment:
This section should be below LogKustoQuery Method .. 
You do want to stop the timeTakeStopWatch and log query before throwing the exception